### PR TITLE
Remove deprecated sonar conf. Try to use the default conf.

### PR DESCRIPTION
**Pull request by jonenst**
Signed-off-by: Jon Harper <jon.harper87@gmail.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Fix coverage report on sonarcloud
- sonar.jacoco.reportPath (merged exec) was deprecated and replaced with sonar.jacoco.reportPaths (multiple execs)
- sonar.jacoco.reportPaths (multiple execs) is deprecated and replaced with sonar.coverage.jacoco.xmlReportPaths (xml reports)

**What is the current behavior?** *(You can also link to an open issue here)*
The coverage report dropped 10% after https://github.com/powsybl/powsybl-core/pull/893 .
This may be because we use a doubly deprecated jacoco-merge.exec system.

**What is the new behavior (if this is a feature change)?**
Try to use the xml reports:
https://sonarcloud.io/documentation/analysis/coverage/

> sonar.jacoco.reportPaths    Deprecated. Use sonar.coverage.jacoco.xmlReportsPath. Path to JaCoCo reports in binary format. Supported only for Java

>  sonar.coverage.jacoco.xmlReportPaths | Comma-delimited list of paths to JaCoCo XML coverage reports.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
NO

### CODE REVIEW

File: distribution-core/pom.xml [hunk 1]
```diff
@@ -462,26 +462,6 @@
             </activation>
             <build>
                 <plugins>
-                    <plugin>
-                        <groupId>org.apache.maven.plugins</groupId>
-                        <artifactId>maven-antrun-plugin</artifactId>
-                        <version>${maven.antrun.version}</version>
```
Comment (mathbagu) [inline distribution-core/pom.xml]: To be confirmed, but it should be possible to remove this property in the parent pom.xml

File: pom.xml [hunk 2]
```diff
@@ -109,11 +109,12 @@
...
 
-        <sonar.jacoco.reportPaths>
-            ../target/jacoco-merge.exec,
-            ../../target/jacoco-merge.exec,
-            ../../../target/jacoco-merge.exec
-        </sonar.jacoco.reportPaths>
+        <sonar.coverage.jacoco.xmlReportPaths>
+            ../distribution-core/target/site/jacoco-aggregate/jacoco.xml,
+            ../../distribution-core/target/site/jacoco-aggregate/jacoco.xml,
+            ../../../distribution-core/target/site/jacoco-aggregate/jacoco.xml
```
Comment (mathbagu) [inline pom.xml]: What is the most robust to a change: `../target/site/jacoco-aggregate/jacoco.xml` vs `../distribution-core/target/site/jacoco-aggregate/jacoco.xml`

File: distribution-core/pom.xml (hunk 1 above)
Comment (jonenst) [reply distribution-core/pom.xml]: removed

File: pom.xml (hunk 2 above)
Comment (jonenst) [reply pom.xml]: As discussed, the report is generated by jacoco:report-aggregate defined in the pom.xml of distribution-core. Therefore by default, the report is generated in distribution-core/target/...

We could change the outputDirectory of the jacoco:report-aggregate to output it in the target of the top project, but we prefer to use the default conf of jacoco:report-aggregate to have less surprise for developers looking for these reports

**Pull request merged**